src/transforms/groupby.js

```diff
@@ -118,7 +128,7 @@ function transformOne(trace, state) {
         }
 
         newTrace.name = groupName;
-        newTrace.marker.color = opts.groupColors[groupName];
+        newTrace = Lib.extendDeep(newTrace, style[groupName] || {});
```

Comment:
I had some false starts but as it turned out, almost all code was already present :-)